### PR TITLE
Update ESP-IDF `c_char` type

### PR DESCRIPTION
**Pull request by SergioGasquez**
# Description

Updates the `c_chart` type for ESP-IDF after https://github.com/rust-lang/rust/pull/132975

# Checklist

- [x] Relevant tests in `libc-test/semver` have been updated
- [x] No placeholder or unstable values like `*LAST` or `*MAX` are
  included (see [#3131](https://github.com/rust-lang/libc/issues/3131))
- [x] Tested locally (`cd libc-test && cargo test --target mytarget`);
  especially relevant for platforms that may not be checked in CI
  - Still not tested, hence the PR is still in draft

### CODE REVIEW

**Review by tgross35 — APPROVED** [review]
Any reason this is marked as a draft?

**Comment by SergioGasquez** [comment]
@ivmarkov wanted to do a small test, that's why I marked it as draft until we confirm that it works, but if you think the change is small enough I can mark it as ready

**Comment by tgross35** [comment]
Happy to have tests! There probably won't be another release until next week anyway.

**Comment by ivmarkov** [comment]
> Happy to have tests! There probably won't be another release until next week anyway.

Tests passed successfully. 
TL;DR: `libc`'s `c_char` and `core::ffi::c_char` should both be `u8`, for all `*-espidf` targets (i.e. both `riscv` as well as `xtensa`).

@SergioGasquez if you could "un-draft" the PR?

Long explanation:
- I checked the Espressif's GCC toolchains (both `xtensa` and `riscv`) and both report the `char` type as **unsigned** by default (`limits.h`'s `CHAR_MAX` is 255).
- I also checked Espressif's `clang` toolchains and there:
  - `clang` with `riscv` also agrees that `CHAR_MAX` is 255 (i.e. **unsigned**)
  - `clang` with `xtensa` still reports `char` as signed, but this is only becausde the Espressif's LLVM+clang `xtensa` fork is not yet updated with the upstream `clang` `xtensa` fix which marks `char` as **unsigned**. There is an open MR to fix the fork to treat `char` as unsigned (just like the upstream code), which will be merged soon

**Comment by tgross35** [comment]
Thanks for confirming!

**Comment by ivmarkov** [comment]
@tgross35 somehow this PR did fall out from the merge queue. Is it something we need to address? Weird.

**Comment by tgross35** [comment]
Thanks for the reminder, I tried adding it back. We just have some tests that are a bit flaky.

**Comment by ivmarkov** [comment]
@tgross35 Sorry to bother - but I think we need this in the 0.2 branch as well.

**Comment by tgross35** [comment]
Indeed, it has the label so it will be in the next release. I was originally planning to do a release next week - but I guess ideally you need it as soon as possible?

Assuming so, I'll try to get one out within the next couple days instead.

**Comment by ivmarkov** [comment]
> Indeed, it has the label so it will be in the next release. I was originally planning to do a release next week - but I guess ideally you need it as soon as possible?
> 
> Assuming so, I'll try to get one out within the next couple days instead.

Yes, ASAP would be deeply appreciated as we are broken since... some time. :-)

Thank you so much!
